mailpoet/tests/integration/Cron/Workers/StatsNotifications/AutomatedEmailsTest.php

```diff
@@ -134,9 +133,15 @@ public function testItDoesntRenderIfNoNewslettersFound() {
   }
 
   public function testItRenders() {
-    $this->createQueue($this->newsletter->getId(), 10);
-    $this->createClicks($this->newsletter->getId(), 5);
-    $this->createOpens($this->newsletter->getId(), 2);
+    $newsletter = $this->newsletterFactory
+      ->withSubject('Subject')
+      ->withWelcomeTypeForSegment(1)
+      ->withActiveStatus()
+      ->withSendingQueue(['count_processed' => 10])
+      ->create();
+
+    $this->createClicks($newsletter, 5);
+    $this->createOpens($newsletter, 2);
```

Comment:
It looks like this part of data preparation is the same in every test case. Shouldn't we move it to the _before function? Or do you prefer having it in each test for better readability?

Reply:
I haven't noticed this repetition when refactoring this part of the code, to be honest.

My first instinct when I saw your question was to say that we should move it to _before() as this is where I believe all setup code should be to make the test methods smaller and easier to read. But then it occurred that it might also make sense to consider in this particular moving this setup code to a method called createNewsletterClicksAndOpens() or something like that and call this method from within the test methods. Just to make it more clear that those specific tests rely on the existence of a newsletter with some clicks and opens while the others don't. Just rambling here, would have to take a closer look to decide which of the two options I prefer.

Happy to hear your thoughts. That is a good question.

Reply:
I like the idea of `createNewsletterClicksAndOpens` method as it solves the code duplicity and its usage in tests more explicitly declares that the test needs those data.

Reply:
@costasovo, since this PR is not merged yet, I went ahead and created the method createNewsletterClicksAndOpens() in c4174b4. @veljkho already approved this PR, and since I only did a small refactor in an integration test, I guess @costasovo can merge it directly after checking the changes that I made, provided there are no issues. But let me know if you prefer a different approach.